tdms/src/iterator.cpp

```diff
@@ -371,1006 +319,577 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
-                               " (" + to_string((int)dimptr_out[0]) +
-                               "," + to_string((int)dimptr_out[1]) +
-                               "," + to_string((int)mxGetNumberOfElements(element)) +
-                               ")\n");
+        if (!(dimptr_out[0] == 1 && dimptr_out[1] == 2)) {
+          throw runtime_error("Incorrect dimension on interface." + field_name + " (" +
+                              to_string((int) dimptr_out[0]) + "," +
+                              to_string((int) dimptr_out[1]) + "," +
+                              to_string((int) mxGetNumberOfElements(element)) + ")\n");
         }
-        if(!strcmp(interface_fields[i],"I0")){
-          place_holder = mxGetPr( (mxArray *)element);
+        if (are_equal(interface_fields[i], "I0")) {
+          place_holder = mxGetPr((mxArray *) element);
           *I0 = *place_holder - 1.;
-          *(I0+1) = *(place_holder+1);
-        }
-        else if(!strcmp(interface_fields[i],"I1")){
-          place_holder = mxGetPr( (mxArray *)element);
+          *(I0 + 1) = *(place_holder + 1);
+        } else if (are_equal(interface_fields[i], "I1")) {
+          place_holder = mxGetPr((mxArray *) element);
           *I1 = *place_holder - 1.;
-          *(I1+1) = *(place_holder+1);
-        }
-        else if(!strcmp(interface_fields[i],"J0")){
-          place_holder = mxGetPr( (mxArray *)element);
+          *(I1 + 1) = *(place_holder + 1);
+        } else if (are_equal(interface_fields[i], "J0")) {
+          place_holder = mxGetPr((mxArray *) element);
           *J0 = *place_holder - 1.;
-          *(J0+1) = *(place_holder+1);
-        }
-        else if(!strcmp(interface_fields[i],"J1")){
-          place_holder = mxGetPr( (mxArray *)element);
+          *(J0 + 1) = *(place_holder + 1);
+        } else if (are_equal(interface_fields[i], "J1")) {
+          place_holder = mxGetPr((mxArray *) element);
           *J1 = *place_holder - 1.;
-          *(J1+1) = *(place_holder+1);
-        }
-        else if(!strcmp(interface_fields[i],"K0")){
-          place_holder = mxGetPr( (mxArray *)element);
+          *(J1 + 1) = *(place_holder + 1);
+        } else if (are_equal(interface_fields[i], "K0")) {
+          place_holder = mxGetPr((mxArray *) element);
           *K0 = *place_holder - 1.;
-          if(*K0<0)
-            *K0 = 0.;
-          *(K0+1) = *(place_holder+1);
-        }
-        else if(!strcmp(interface_fields[i],"K1")){
-          place_holder = mxGetPr( (mxArray *)element);
+          if (*K0 < 0) *K0 = 0.;
+          *(K0 + 1) = *(place_holder + 1);
+        } else if (are_equal(interface_fields[i], "K1")) {
+          place_holder = mxGetPr((mxArray *) element);
           *K1 = *place_holder - 1.;
-          if(*K1<0)
-            *K1 = 0.;
-          *(K1+1) = *(place_holder+1);
-        }
-        else{
-          throw runtime_error("element interface."+field_name+" not handled");
+          if (*K1 < 0) *K1 = 0.;

```

Comment:
This silently ignores if K is not a viable index in the input and sets it to zero. Maybe something should be thrown here?